src/streams.c

```diff
@@ -779,27 +779,13 @@ static Obj PRINT_OR_APPEND_TO(Obj args, int append)
 
     /* first entry is the filename                                         */
     filename = ELM_LIST(args,1);
-    while ( ! IsStringConv(filename) ) {
-        filename = ErrorReturnObj(
-            "%s: <filename> must be a string (not a %s)",
-            (Int)funcname, (Int)TNAM_OBJ(filename),
-            "you can replace <filename> via 'return <filename>;'" );
-    }
```

Comment:
You write that `PrintTo` already checks that `filename` is a string. But in `lib/streams.gd:1044` only `IsString` is checked -- but that is not enough, it really needs to be in `IsStringRep`. So you risk a crash there.